src/hooks/useCheckImage.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from "react";
+import { useAsyncFn } from "react-use";
+
+export const useCheckImage = (
+  src: string
+): {
+  isValid: boolean;
+  isLoading: boolean;
+  error: string | undefined;
+} => {
+  const [{ loading, error }, checkImage] = useAsyncFn(async () => {
```

Comment:
If you change `useAsyncFn` to `useAsync` you can get rid of the `useEffect` and and logic will run the same way.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1995/commits/5d9989860f08448b0260105f0c054f519b00bd03